org.eclipse.gef/src/org/eclipse/gef/tools/MarqueeSelectionTool.java

```diff
@@ -303,16 +301,10 @@ private Collection calculatePrimaryMarqueeSelectedEditParts() {
 					.getAllNestedConnectionEditParts((GraphicalEditPart) getCurrentViewer().getRootEditPart()));
 		}
 
-		// process all edit parts and determine which are affected by the
-		// current marquee selection
-		Collection marqueeSelectedEditParts = new ArrayList();
-		for (Object element : editPartsToProcess) {
-			GraphicalEditPart editPart = (GraphicalEditPart) element;
-			if (isMarqueeSelectable(editPart) && isPrimaryMarqueeSelectedEditPart(editPart)) {
-				marqueeSelectedEditParts.add(editPart);
-			}
-		}
-		return marqueeSelectedEditParts;
+		// process all edit parts and determine which are affected by the current
+		// marquee selection
+		return editPartsToProcess.stream()
+				.filter(gep -> (isMarqueeSelectable(gep) && isPrimaryMarqueeSelectedEditPart(gep))).toList();
```

Comment:
I would split this up into two "filter" calls:
`filter(this::isMarqueeSelectable).filter(this::isPrimaryMarqueeSelectEditPart)`

Reply:
you are right this is much nicer. changed.